src/runtime_src/core/tools/xbflash2/OO_Dump_Qspips.cpp

```diff
@@ -0,0 +1,198 @@
+                std::stringstream sstream(slength);
+                sstream >> len;
+            }
+        }
+    }
+    catch (...) {
+    }
+
+    //Optional arguments
+    try {
+        if (vm.count("bar")) {
+            sBar = vm["bar"].as<std::string>();
+            if (!sBar.empty())
+                bar = std::stoi(sBar);
+        }
+        if (vm.count("bar-offset")) {
+            sBarOffset = vm["bar-offset"].as<std::string>();
+            if (!sBarOffset.empty()) {
+                std::stringstream sstream(sBarOffset);
+                sstream >> baroff;
+            }
+        }
+    }
+    catch (...) {
+    }
+
+    if (bdf.empty() || output.empty())
+        throw std::errc::invalid_argument;
+
+    std::cout << "Read out flash"
+        << boost::format("[0x%x, 0x%x] on device %s to %s\n") % offset % (offset + len) % bdf % output;
+
+    pcidev::pci_device dev(bdf, bar, baroff, flash_type);
+    XQSPIPS_Flasher qspips(&dev);
+
+    return qspips.xclReadBack(output, offset, len);
+}
+} //end namespace 
+
+// ----- C L A S S   M E T H O D S -------------------------------------------
+
+OO_Dump_Qspips::OO_Dump_Qspips( const std::string &_longName, bool _isHidden )
+    : OptionOptions(_longName, _isHidden, "Flash type - qspips")
+    , m_help(false)
+{
+  m_optionsDescription.add_options()
+      ("device,d", po::value<std::string>(), "Bus:Device.Function\" (e.g., 0000:d8:00.0) device of interest.\n")
+      ("offset,a", po::value<std::string>(), "offset on flash to start, default is 0.\n")
+      ("length,l", po::value<std::string>(), "length-to-read, default is 128MB.\n")
+      ("flash-part,p", po::value<std::string>(), "qspips-flash-part, default is qspi_ps_x2_single.\n")
+      ("bar,b", po::value<std::string>(), "BAR-index-for-QSPIPS, default is 0.\n")
+      ("bar-offset,s", po::value<std::string>(), "BAR-offset-for-QSPIPS, default is 0x40000.\n")
+      ("output,o", po::value<std::string>(), "output file to save read contents.\n")
+      ("help", boost::program_options::bool_switch(&m_help), "Help to use this sub-command")
+  ;
+}
+
+void
+OO_Dump_Qspips::execute(const SubCmdOptions& _options) const
+{
+  XBU::verbose("SubCommand dump - option: Flash type - qspips");
+
+  XBU::verbose("Option(s):");
+  for (auto & aString : _options)
+      XBU::verbose(std::string(" ") + aString);
+
+  // Honor help option first
+  if (std::find(_options.begin(), _options.end(), "--help") != _options.end()) {
+    printHelp();
+    return;
+  }
+
+  // Parse sub-command ...
+  po::variables_map vm;
+
+  try {
+    po::store(po::command_line_parser(_options).options(m_optionsDescription).run(), vm);
+    po::notify(vm); // Can throw
+  }
+  catch (po::error& e) {
+    std::cerr << "ERROR: " << e.what() << "\n\n";
+    printHelp();
+    throw std::errc::operation_canceled;
+  }
+
+  if(m_help) {
+    printHelp();
+    throw std::errc::operation_canceled;
+  }
+
+  XBU::sudo_or_throw_err();
+
+  try {
+      if (qspips_readback(vm)) {
```

Comment:
Does qspips_readback() throw and exception.  If so remove the if statement if not remove the try/catch block.